react/features/mobile/external-api/middleware.js

```diff
@@ -143,6 +179,28 @@ function _sendConferenceEvent(
         || _sendEvent(store, _getSymbolDescription(type), data);
 }
 
+/**
+ * Sends {@link CONFERENCE_FAILED} event when the {@link CONNECTION_FAILED}
+ * occurs. Otherwise the external API will not emit such event, because at this
+ * point conference has not been created yet and the base/conference feature
+ * will not emit it.
+ *
+ * @param {Store} store - The redux store.
+ * @param {Action} action - The redux action.
+ * @returns {void}
+ */
+function _sendConferenceFailedOnConnectionError(store, action) {
+    const { locationURL } = store.getState()['features/base/connection'];
```

Comment:
Just curious: What's the (is there any) difference between the locationURL retrieved from base/connection and the conference info retrieved from action.conference? Can it cause any issues that we use this instead of the one in the action?

Reply:
hmm I don't see the locationURL to be part of the connectionFailed action:
https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/connection/actions.native.js#L188

The locationURL is set as part of the setLocation action:
https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/connection/actions.native.js#L322

And the location is set, before new connection instance is created. I've made an assumption that the locationURL is up to date for the current connection instance.